cmd/virtual-kubelet/main.go

```diff
@@ -16,71 +16,302 @@ package main
+
+		logLevel        = "info"
+		traceSampleRate string
+
+		// for aci
+		kubeconfigPath  = os.Getenv("KUBECONFIG")
+		cfgPath         string
+		clusterDomain   = "cluster.local"
+		startupTimeout  time.Duration
+		disableTaint    bool
+		operatingSystem = "Linux"
+		numberOfWorkers = 50
+		resync          time.Duration
+
+		certPath       = os.Getenv("APISERVER_CERT_LOCATION")
+		keyPath        = os.Getenv("APISERVER_KEY_LOCATION")
+		clientCACert   = os.Getenv("APISERVER_CA_CERT_LOCATION")
+		clientNoVerify bool
+
+		webhookAuth                  bool
+		webhookAuthnCacheTTL         time.Duration
+		webhookAuthzUnauthedCacheTTL time.Duration
+		webhookAuthzAuthedCacheTTL   time.Duration
+		nodeName                     string
+		listenPort                   = 10250
+
+		// deprecated
+		namespace   string
+		metricsAddr string
+		provider    string
+		leases      bool
+	)
+
+	if kubeconfigPath == "" {
+		home, _ := homedir.Dir()
+		if home != "" {
+			kubeconfigPath = filepath.Join(home, ".kube", "config")
+		}
+	}
 
-	logger := logrus.StandardLogger()
-	log.L = logruslogger.FromLogrus(logrus.NewEntry(logger))
-	logConfig := &logruscli.Config{LogLevel: "info"}
+	withTaint := func(cfg *nodeutil.NodeConfig) error {
+		if disableTaint {
+			return nil
+		}
 
-	trace.T = opencensus.Adapter{}
-	traceConfig := opencensuscli.Config{
-		AvailableExporters: map[string]opencensuscli.ExporterInitFunc{
-			"ocagent": initOCAgent,
+		taint := v1.Taint{
+			Key:   taintKey,
+			Value: taintValue,
+		}
+		switch taintEffect {
+		case "NoSchedule":
+			taint.Effect = v1.TaintEffectNoSchedule
+		case "NoExecute":
+			taint.Effect = v1.TaintEffectNoExecute
+		case "PreferNoSchedule":
+			taint.Effect = v1.TaintEffectPreferNoSchedule
+		default:
+			return errdefs.InvalidInputf("taint effect %q is not supported", taintEffect)
+		}
+		cfg.NodeSpec.Spec.Taints = append(cfg.NodeSpec.Spec.Taints, taint)
+		return nil
+	}
+	withVersion := func(cfg *nodeutil.NodeConfig) error {
+		cfg.NodeSpec.Status.NodeInfo.KubeletVersion = strings.Join([]string{k8sVersion, "vk-azure-aci", buildVersion}, "-")
+		return nil
+	}
+
+	configureRoutes := func(cfg *nodeutil.NodeConfig) error {
+		mux := http.NewServeMux()
+		cfg.Handler = mux
+		return nodeutil.AttachProviderRoutes(mux)(cfg)
+	}
+	withWebhookAuth := func(cfg *nodeutil.NodeConfig) error {
+		if !webhookAuth {
+			return nil
+		}
+		auth, err := nodeutil.WebhookAuth(cfg.Client, cfg.NodeSpec.Name, func(cfg *nodeutil.WebhookAuthConfig) error {
+			if webhookAuthnCacheTTL > 0 {
+				cfg.AuthnConfig.CacheTTL = webhookAuthnCacheTTL
+			}
+			if webhookAuthzAuthedCacheTTL > 0 {
+				cfg.AuthzConfig.AllowCacheTTL = webhookAuthzAuthedCacheTTL
+			}
+			if webhookAuthzUnauthedCacheTTL > 0 {
+				cfg.AuthzConfig.AllowCacheTTL = webhookAuthzUnauthedCacheTTL
+			}
+
+			if clientCACert != "" {
+				ca, err := dynamiccertificates.NewDynamicCAContentFromFile("client-ca", clientCACert)
```

Comment:
The client certificate in Virtual Kueblet named `ca-cert-bundle` as shown [here](https://github.com/virtual-kubelet/virtual-kubelet/blob/7c9bd20eea6f4b37929fb765ca80d69262ca51c5/cmd/virtual-kubelet/internal/commands/root/root.go#L186). Any problem with having different file name?